lib/rubocop/cop/internal_affairs/redundant_location_argument.rb

```diff
@@ -10,36 +10,48 @@ module InternalAffairs
       # @example
       #
       #   # bad
-      #   add_offense(node, :expression)
+      #   add_offense(node, location: :expression)
       #
       #   # good
       #   add_offense(node)
-      #   add_offense(node, :selector)
-      #   add_offense(node, :expression, 'message')
+      #   add_offense(node, location: :selector)
       #
       class RedundantLocationArgument < Cop
         MSG = 'Redundant location argument to `#add_offense`.'.freeze
 
-        def_node_matcher :node_type_check, <<-PATTERN
-          (send nil :add_offense _ (sym :expression))
+        def_node_matcher :add_offense_kwargs, <<-PATTERN
+          (send nil :add_offense _ $hash)
+        PATTERN
+
+        def_node_matcher :location_expression_argument?, <<-PATTERN
+          (pair (sym :location) (sym :expression))
         PATTERN
 
         def on_send(node)
-          node_type_check(node) do
-            add_offense(node.last_argument)
-          end
+          offending_node = location_expression_argument(node)
```

Comment:
I would make `#location_expression_argument` yield anything it finds, and `#add_offense` in a block.